drivers/soundwire/intel_init.c

```diff
@@ -244,6 +244,11 @@ static struct sdw_intel_ctx
 
 		link->cdns = dev_get_drvdata(&ldev->auxdev.dev);
 
+		if (!link->cdns) {
+			dev_err(&adev->dev, "failed to get link->cdns\n");
+			auxiliary_device_uninit(&ldev->auxdev);
```

Comment:
I think either you have to do auxiliary_device_delete() and uninit() in this case or you can simply call intel_link_dev_unregister() here right?

Reply:
> I think either you have to do auxiliary_device_delete() and uninit() in this case or you can simply call intel_link_dev_unregister() here right?

Thanks, update it now.